retrier.go

```diff
@@ -0,0 +1,36 @@
+package retry
+
+import (
+	"context"
+	"time"
+)
+
+// Retrier implements an exponentially backing off retry instance.
+// Use New instead of creating this object directly.
+type Retrier struct {
+	delay       time.Duration
+	floor, ceil time.Duration
+}
+
+// New creates a retrier that exponentially backs off from floor to ceil pauses.
+func New(floor, ceil time.Duration) *Retrier {
+	return &Retrier{
+		delay: floor,
+		floor: floor,
+		ceil:  ceil,
+	}
+}
+
+func (r *Retrier) Wait(ctx context.Context) bool {
+	const growth = 2
+	r.delay *= growth
+	if r.delay > r.ceil {
+		r.delay = r.ceil
+	}
+	select {
+	case <-time.After(r.delay):
```

Comment:
Might be a good idea to use time.NewTimer here to avoid memory usage when things are failing consistently and contexts are being cancelled.